src/backend/buffer.rs

```diff
@@ -158,11 +167,9 @@ where
 
     #[cfg_attr(feature = "inline-more", inline)]
     fn with_capacity(capacity: usize) -> Self {
-        /// We encode the `usize` string length into the buffer as well.
-        const LEN_USIZE: usize = mem::size_of::<usize>();
-        /// According to google the approx. word length is 5.
-        const DEFAULT_STR_LEN: usize = 5;
-        let bytes_per_string = DEFAULT_STR_LEN + LEN_USIZE;
+        // We encode the `usize` string length into the buffer as well.
+        let var7_len: usize = calculate_var7_size(capacity);
+        let bytes_per_string = DEFAULT_STR_LEN + var7_len;
```

Comment:
As said in a prior comment, we should probably get rid of all the heuristics based `with_capacity` methods and get this down to the basics. A Rust `BTreeMap` for example also does not provide a `with_capacity` method because it simply does not make sense to do so.

Reply:
I agree, it's rather pointless.